src/main/java/org/ballerinalang/command/cmd/ListCommand.java

```diff
@@ -91,31 +97,71 @@ public void setParentCmdParser(CommandLine parentCmdParser) {
      * @param outStream stream outputs need to be printed
      */
     private static void listDistributions(PrintStream outStream) {
+        String currentBallerinaVersion = ToolUtil.getCurrentBallerinaVersion();
         try {
-            String currentBallerinaVersion = ToolUtil.getCurrentBallerinaVersion();
             File folder = new File(ToolUtil.getDistributionsPath());
             File[] listOfFiles;
             listOfFiles = folder.listFiles();
+            Arrays.sort(listOfFiles);
+            List<Channel> channels = ToolUtil.getDistributions();
             outStream.println("Distributions available locally: \n");
             List<String> installedVersions = new ArrayList<>();
-            for (int i = 0; i < listOfFiles.length; i++) {
-                if (listOfFiles[i].isDirectory()) {
-                    String version = listOfFiles[i].getName().split("-")[1];
-                    outStream.println(markVersion(currentBallerinaVersion, version)
-                            + " " +  ToolUtil.getType(version) + " version " + version);
-                    installedVersions.add(version);
+            for (Channel channel : channels) {
+                for (Distribution distribution : channel.getDistributions()) {
+                    for (int i = 0; i < listOfFiles.length; i++) {
+                        if (listOfFiles[i].isDirectory()) {
+                            String version = listOfFiles[i].getName().split("-")[1];
+                            if (version.equals(distribution.getVersion())) {
+                                outStream.println(markVersion(currentBallerinaVersion, version)
+                                        + " " + ToolUtil.getType(version));
+                                installedVersions.add(version);
+                            }
+                        }
+                    }
                 }
             }
-            outStream.println("\nDistributions available remotely:");
-            List<Channel> channels = ToolUtil.getDistributions();
+            FileWriter writer = null;
+            try {
+                writer = new FileWriter("localDists.json");
```

Comment:
Since its a file name shall we use - or _ to separate instead of using pascal case?
Also declare this name as a constant